src/main/java/com/example/user/domain/User.java

```diff
@@ -0,0 +1,49 @@
+package com.example.user.domain;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.Pattern;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@Entity
+public class User extends BaseEntity {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  @Column(name ="id",columnDefinition = "BIGINT")
+  private Long id;
+
+  @Pattern(regexp = "\\b[\\w\\.-]+@[\\w\\.-]+\\.\\w{2,4}\\b")
+  @Column(name = "email", columnDefinition = "VARCHAR(100)", nullable = false, unique = true)
+  private String email;
+
+  @Column(name = "username", columnDefinition = "VARCHAR(50)", nullable = false)
+  private String username;
+
+  @Column(name = "password", columnDefinition = "VARCHAR(100)", nullable = false)
+  private String password;
+
+  @Column(name = "nickname", columnDefinition = "VARCHAR(50)", unique = true)
```

Comment:
columnDefinition을 통해 길이를 잡으시지 마시고 length 프로퍼티를 사용하시면 됩니다!
문자열을 통한 설정은 오타의 위험이 있기에 columnDefinition을 통한 칼럼 설정은 지향하고 있습니다.

```suggestion
  @Column(name = "nickname", length=50, unique = true)
```